addons/folo/common/src/main/java/org/commonjava/indy/folo/data/FoloProducer.java

```diff
@@ -0,0 +1,29 @@
+package org.commonjava.indy.folo.data;
+
+import org.commonjava.indy.folo.conf.FoloConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+
+
+@ApplicationScoped
+public class FoloProducer {
+
+    private final Logger logger = LoggerFactory.getLogger( this.getClass() );
+
+    @Inject
+    FoloConfig foloConfig;
+
+
+    @Produces
+    @ApplicationScoped
+    public FoloRecord getFoloRecordCassandra(@FoloStoreToCassandra FoloRecord dbRecord
+            ,@FoloStoretoInfinispan FoloRecord cacheRecord) {
+        return foloConfig.getStoreToCassandra() ? dbRecord : cacheRecord;
```

Comment:
IMO we should use the IndyConfiguration.isStandalone() flag for this instead, so we can switch all of this on or off from one flag, and avoid situations where we have data stored in surprising places due to partial configuration.

Also, do we need to worry about the Infinispan (or Cassandra) implementation of this interface being unable to construct / inject in cases where we're not using it? DefaultGalleyStorageProvider can serve as another example we could follow, if we need more handling to reduce init-time problems.

Reply:
I agree on  "...unable to construct / inject... ". we should not inject both and pick up one. We should only construct one.

Reply:
Using a lot of different configuration on onee same config value. is much. worse then using specific values for specific. purposes. 
IMHO ff for some. reason. we. change that value (which we are using for several options. switching on/off). then we. will. experience unwanted result and. we. will. once. again need to return back. and find. every. single place where. this was used and change or provide other implementation.
BTW in openshift we are using configmap for. configuration so. we don't. need to worry about partial stored config values.

No we don't need to worry because this will. inject right. instance for. right purpose. If. we would. like to. use. infinispan. cache. then folo.cassandra=false will. produce. cache instance for storing data in cache and the other way around for producing cassandra storage if folo.cassandra=true. ( BTW this is from weld. documentation for injecting right instance
link: https://docs.jboss.org/weld/reference/2.4.6.Final/en-US/pdf/weld-reference.pdf -  page 29 )